packaging/tags.py

```diff
@@ -231,16 +241,13 @@ def cpython_tags(
     for abi in abis:
         for platform_ in platforms:
             yield Tag(interpreter, abi, platform_)
-    # Not worrying about the case of Python 3.2 or older being specified and
-    # thus having redundant tags thanks to the abi3 in-fill later on as
-    # 'packaging' doesn't directly support Python that far back.
-    if len(python_version) > 1:
+    if _abi3_applies(python_version):
         for tag in (Tag(interpreter, "abi3", platform_) for platform_ in platforms):
             yield tag
     for tag in (Tag(interpreter, "none", platform_) for platform_ in platforms):
         yield tag
-    # PEP 384 was first implemented in Python 3.2.
-    if len(python_version) > 1:
+
+    if _abi3_applies(python_version):
         for minor_version in range(python_version[1] - 1, 1, -1):
```

Comment:
This will result in a few nonsense tags being added (`cp31-abi3-<platform>`, `cp30-abi3-<platform>`).

Reply:
Actually it won't due to `range(...,1, ...)`; you can't hit that block unless you're Python 3.2 or newer, and if you're Python 3.2 then that's `range(1, 1, -1)` which is empty due to `range()` being exclusive with it's end value.

I've added extra tests for Python 3.3 and 3.2 to make double-sure this isn't an issue. 😄

Reply:
Ah, I switched end and step when reading. Thanks. :)